### ISSUE TITLE
Missing python lldb type hints for mypy

### ISSUE THREAD

**Issue by briandw** [llvm/llvm-project#79043]
echo "import lldb" > test.py
mypy test.py 
test.py:1: error: Skipping analyzing "lldb": module is installed, but missing library stubs or py.typed marker  [import-untyped]
test.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)

Seems like the model has type hints. Why doesn't the module support mypy?

**Comment by kastiglione** [comment]
> Seems like the model has type hints.

If you're looking at lldb.py, there are comments that look like type hints, but they're invalid. These comments are generated by [SWIG](https://www.swig.org). The main problem is that the comments reference types by their C++ name, ex: `const char *`, or `lldb::addr_t`, etc.

**Comment by porglezomp** [comment]
It looks like the SWIG-side issue is https://github.com/swig/swig/issues/735

**Comment by Nerixyz** [comment]
Hi, I wanted to give an update on this. I opened some PRs for SWIG ([PRs](https://github.com/swig/swig/pulls?q=sort%3Aupdated-desc+is%3Apr+author%3ANerixyz+)) to get basic type annotations working.

Still waiting on a review for some of them, and I've got some more improvements/features prepared. Though, https://redirect.github.com/swig/swig/pull/3390 is the most important one that gets us 90% there, as it shows the class name in type annotations. 

I have all my progress in https://github.com/Nerixyz/swig/tree/feat/python-pep484-all right now (hoping to get all these into SWIG 4.5, some parts need polishing+docs). To test it, I enabled the annotations to LLDB here: https://github.com/Nerixyz/llvm-project/tree/feat/lldb-pytyping. You can check out the produced Python stubs here: https://gist.github.com/Nerixyz/56b240aa39238ce8a26bb56152c5f9a6. As you can see, there are only seven "special" incomplete classes that SWIG couldn't resolve (at the bottom). From glancing over the annotations, most seem reasonable. Much better than the types from the docstrings.

If you have feedback on the general design, now is the best time, as this isn't set in stone yet.

---

PS: If anyone wants to hack on SWIG in the future, I put together some printers for their `DOH*` types: https://gist.github.com/Nerixyz/59791150ef35d492826e705f936d739a. They only cover the basics, but that was good enough for me.

**Comment by k4lizen** [comment]
thanks for working on this!

**Comment by Nerixyz** [comment]
Update: A lot of my Swig PRs have been merged into Swig 4.5, and I enabled the annotations here in #213463. There are some shortcomings mentioned in the PR, but the majority of functions/methods are annotated. If you find other issues, please report them.
